web/docs/auth/entities/entities.md

```diff
@@ -1,22 +1,454 @@
+
+Before accessing some auth method's data, you'll need to check if that data exists for the user and then access it:
+
+```ts
+if (user.identities.google !== null) {
+  const userId = user.identities.google.id
+  // ...
+}
+```
+
+You need to do this because if a user didn't sign up with some auth method, the data for that auth method will be `null`.
+:::
+
+Let's look at the data for each of the available auth methods:
+
+- [Username & password](../username-and-pass.md) data
+
+  <UsernameData />
+
+- [Email](../email.md) data
+
+  <EmailData />
+
+- [Google](../social-auth/google.md) data
+
+  <GoogleData />
+
+- [GitHub](../social-auth/github.md) data
+
+  <GithubData />
+
+- [Keycloak](../social-auth/keycloak.md) data
+
+  <KeycloakData />
+
+If you support multiple auth methods, you'll need to find which identity exists for the user and then access its data:
+
+```ts
+if (user.identities.email !== null) {
+  const email = user.identities.email.id
+  // ...
+} else if (user.identities.google !== null) {
+  const googleId = user.identities.google.id
+  // ...
+}
+```
+
+### `getFirstProviderUserId` Helper
+
+The `getFirstProviderUserId` method returns the first user ID (e.g. `username` or `email`) that it finds for the user or `null` if it doesn't find any.
+
+[As you'll learn below](#authidentity-entity-), the `providerUserId` field is how providers identify our users. For example, the user's `username` in the case of the username auth or the user's `email` in the case of the email auth. This can be useful if you support multiple authentication methods and you need _any_ ID that identifies the user in your app.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx title="src/MainPage.jsx"
+const MainPage = ({ user }) => {
+  const userId = user.getFirstProviderUserId()
+  // ...
+}
+```
+
+```js title=src/tasks.js
+export const createTask = async (args, context) => {
+  const userId = context.user.getFirstProviderUserId()
+  // ...
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="src/MainPage.tsx"
+import { type AuthUser } from 'wasp/auth'
+
+const MainPage = ({ user }: { user: AuthUser }) => {
+  const userId = user.getFirstProviderUserId()
+  // ...
+}
+```
+
+```ts title=src/tasks.ts
+export const createTask: CreateTask<...>  = async (args, context) => {
+  const userId = context.user.getFirstProviderUserId()
+  // ...
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Including the User with Other Entities
+
+Sometimes, you might want to include the user's data when fetching other entities. For example, you might want to include the user's data with the tasks they have created.
+
+:::caution Be careful to exclude the password
+
+You'll need to include the `auth` and the `identities` relations to get the full data about the user. However, you should exclude the `providerData` field from the `AuthIdentity` entity because it contains the user's hashed password.
```

Comment:
The phrasing `might want to include to get the full auth data` sounds lean and strict at the same time to me 😄 

They _need_ to include it to get the auth data, so I've kept the `need` part, but I've used your version to modify the text.

Reply:
I guess it wasn't so clear to me what auth data means. Or I wanted to make it clear they might not need the full auth data. Because often they won't, they will save some stuff on `user` directly instead.
Ok, good!